feat: More robust multi-threaded speed testing with more generalized provider support

mitchcapper opened the pull request:
New Fast.com provider is not exposed to the user currently, but once options are added it can be.  This does a better job of evaluating high speed connections and follows more closely to official testers behavior.

I have imported some of my own code here, so if it looks like overkill that is likely why.  This gets me very reliable results over  many tests.   It should also scale the test to the size of the connection, like the actual tests.

No longer directly sub-classing HttpClient as I didn't see the benefit of going that route.

XeroxDev commented on SpeedtestPlugin/Classes/ByteBitSize.cs:
```diff
@@ -0,0 +1,110 @@
+    }
+
+}
+namespace _IgnoreIgnoreIgnore
```
I'm not a bug friend of such namespaces. Is there any reason for this? If no, could you please change this?

XeroxDev reviewed (changes requested):
Expect the one thing, seems extremely good! Nice work!

XeroxDev commented:
I'll merge it, when I'm doing a release.

XeroxDev reviewed (changes requested):
I've tested your changes and I've to inform you, that those changes are breaking the plugins functionality.

1. you're using "min_ping" to display the ping. For me it's always 0.
2. Upload throws exceptions. See https://p.tswi.me/bojojepoxa.exception for full stack trace.
3. Download is still inaccurate. Plugin displays for me 42.8 MB/s but if I use the service you've implemented (fast.com / from netflix) I get 970Mb/s a.k.a ~120MB/s

mitchcapper commented:
Try again if still an issue please send the log:)

mitchcapper committed:
feat: More robust multi-threaded speed testing with more generalized provider support

New Fast.com provider is not exposed to the user currently, but once options are added it can be.  This does a better job of evaluating high speed connections and follows more closely to official testers behavior.
